implementations/micrometer-registry-otlp/src/main/java/io/micrometer/registry/otlp/OtlpConfig.java

```diff
@@ -228,18 +229,40 @@ default HistogramFlavor histogramFlavor() {
     }
 
     /**
-     * Configures the histogram flavor to use on a per-meter level. This will override the
-     * {@link #histogramFlavor()} configuration for matching Meters. The key is used to do
-     * an exact match on the Meter's name.
+     * Configures the histogram flavor mapping to use on a per-meter level. This can
+     * override the {@link #histogramFlavor()} configuration for matching Meters. By
+     * default, {@link #histogramFlavorPerMeter(Meter.Id)} uses the result of this method
+     * to look up the {@link HistogramFlavor} by {@link Meter.Id} (exact match on the
+     * Meter's name by default). This means that this method provides the data while
+     * {@link #histogramFlavorPerMeter(Meter.Id)} provides the logic for the lookup, and
+     * you can override them independently.
      * @return mapping of meter name to histogram flavor
      * @since 1.15.0
+     * @see #histogramFlavorPerMeter(Meter.Id)
      * @see #histogramFlavor()
      */
     default Map<String, HistogramFlavor> histogramFlavorPerMeter() {
         return getStringMap(this, "histogramFlavorPerMeter", HistogramFlavor::fromString)
             .orElse(Collections.emptyMap());
     }
 
+    /**
+     * Looks up the histogram flavor to use on a per-meter level. This will override the
+     * {@link #histogramFlavor()} lookup behavior for matching Meters. By default, the key
+     * is used to do an exact match on the Meter's name in the map that
+     * {@link #histogramFlavorPerMeter()} returns. This means that
+     * {@link #histogramFlavorPerMeter()} provides the data while this method provides the
+     * logic for the lookup, and you can override them independently.
+     * @param id the {@link Meter.Id} the {@link HistogramFlavor} is configured for
+     * @return the histogram flavor mapped to the {@link Meter.Id}
+     * @since 1.15.0
+     * @see #histogramFlavorPerMeter()
+     * @see #histogramFlavor()
+     */
+    default HistogramFlavor histogramFlavorPerMeter(Meter.Id id) {
+        return histogramFlavorPerMeter().getOrDefault(id.getName(), histogramFlavor());
```

Comment:
As mentioned offline, I think we should make a type for the function that is this method's signature and make it configurable on the registry Builder rather than in the OtlpConfig. The reason is to keep logic in the registry and config in the config interface - this feels a bit too much like logic for how the config is used. Probably it would be most usable for most people for the default implementation to copy the longest starts-with match wins behavior they have in Spring Boot for some similar types of configuration properties. If that were the case, I suspect the vast majority of users would not want to customize the logic anyway, so the fact it needs to be done via the builder is not much of an issue. Thoughts?